src/CalculatorW2.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file CalculatorW2.cpp
+ * @author your name (you@domain.com)
+ * @brief 
+ * @version 0.1
+ * @date 2021-07-20
+ * 
+ * @copyright Copyright (c) 2021
+ * 
+ */
+
+#include <float.h>
+#include "CalculatorW2.h"
+
+namespace calculatorw2
+{
+    CalculatorW2::CalculatorW2(): error_(nullptr)
+    {
+
+    }
+    
+    CalculatorW2::~CalculatorW2() {}
+    
+    const char* CalculatorW2::getError()
+    {
+        return error_;
+    }
+
+    double CalculatorW2::calculate(double left, double right, char operation)
+    {
+        double result = 0;
+        
+        switch (operation)
+        {
+            case '+':
```

Comment:
consider using your enum from CharacterStrings here, you may need to cast, e.g. `(char)Operators::ADD`